### PR TITLE
2FA Backends

### PR DESCRIPTION
**Pull request by protich**
This PR rewrites and  borrows from work done by @aydreeihn in PR #5538. Main highlights being -  adding  true 2FA Backends as opposed to piggybacking on Authentication Backends and smoother integration to the login process.

TODO:

- [x] Add email template for email-2fa to Admin Panel > Setting > Agents > Templates
- [x] Add tooltips as needed
- [x] Add docs on how the feature works
- [x] Bug fixes

### CODE REVIEW

**Comment by TechCare-Aus** [comment]
@protich you sir are a legend. Very excited to have this implemented.

**Comment by TechCare-Aus** [comment]
With the recent security scares guys, I'm really hoping that this will become a huge priority.
I've noticed a lot of good improvements happening, so really looking forward to this being a thing soon.

**Comment by TechCare-Aus** [comment]
@protich is this function only going to be supported in the upcoming osTicket? I noticed that it is being applied to develop-next and not develop.

**Comment by JediKev** [comment]
@TechCare-Aus 

`develop-next` is our "development" branch where we merge new features for the next major release. Essentially, you can think of it as the testing branch for the next major release (v1.15 - or whatever we decide to name it in this case). This helps us keep `develop` clean and stable while we test out new major features and so on.

TLDR;
This will be included in a future release.

Cheers.

**Comment by TechCare-Aus** [comment]
@JediKev - Thanks for the heads up mate. In relation then to develop-next, does this include all the commits to develop? I noticed that develop-next shows:

image

Which I assume means that the various commits that develop has received to fix stability bugs have not been applied to develop-next. I would have thought that develop-next would include this as well?

**Comment by JediKev** [comment]
@TechCare-Aus 

Yea, they will be merged eventually we just haven't gotten around to it yet. Trust me, we won't forget them lol

Cheers.

**Comment by TechCare-Aus** [comment]
@JediKev - no drama mate. Thanks for letting us know. Looking forward to the next release. 2FA is huge now.
